File: arrays/repeat-and-missing-number-array.py
class Solution:
    def repeatedNumber(self, A):
        list_n_sum = sum(A)
        list_sq_sum = sum(map(lambda x: x*x, A))

        n = len(A)
        n_sum = n*(n+1)/2
        sq_sum = n*(n+1)*(2*n+1)/6

        x = list_n_sum - n_sum # = a - b, where a is double & b is missing
        y = list_sq_sum - sq_sum # = a^2 - b^2

        a = (x + y/x)/2
        b = a - x

        return [a, b]

File: arrays/repeat-and-missing-number-array.cpp
vector<int> Solution::repeatedNumber(const vector<int> &A) {
    long long int i=0, j=0, k=0, r, s, n = A.size(), x, y;
    r = n*(n+1)/2;
    s = n*(n+1)*(2*n+1)/6;
    
    for(i=0; i<n; i++){
        j+=A[i];
        k+=(long long int)A[i]*(long long int)A[i];
    }
    
    x = r-j;
    y = s-k;

    vector<int> ans;
    ans.push_back(((y/x)-x)/2);
    ans.push_back(x+ans[0]);
    return ans;
}


